### DISCUSSION TITLE
Backwards compatibility for older windows versions (WinXP)

### DISCUSSION THREAD

**Discussion by Aderinom** [microsoft/msquic#2283] [Q&A] [answered]
Hello Everyone! 

I'm currently evaluating msquic for a project in our company.
One of the requirements however will be compatibility for 32bit Windows XP.

I am aware that this is not a listed supported platform, also that MSVC2019  doesn't support WinXP as a build target anymore. 

However I'd be interested if anyone has some experience/tricks how this might still be achievable.

**Comment by nibanks** [comment, marked as answer]
I'm not sure. I know it would require some significant work, since we use some APIs that don't exist before Windows 10, but it's possible they might be stubbed out or replaced. But I don't have any experience on compiling for XP any more and it's definitely not a goal of ours to support that. We can help with any specific questions you might have if you attempt to get it building.

**Reply by ThadHouse** [reply]
XP was before the networking stack was rewritten in Vista, so I imagine there will need to be some major changes in the datapath code, especially around the socket per processor code.

Additionally, you will be forced to use openssl, which means you'll also have to figure out how to compile a new version of openssl for XP, since you need a new version to have quic support.

**Reply by Aderinom** [reply]
Thank you both for the quick reply!

So XP will be hard to get running. 
What is the latest windows version which is currently in scope for MSquic? 
From @nibanks answer I'd assume Win10?

**Reply by nibanks** [reply]
Correct. Win 10 and Win 11 are supported.
